### PR TITLE
Adding unexposed method to public contract

### PR DESCRIPTION
**Pull request by geleems**
I added more unexposed methods, which was implemented already, to public contract.

### CODE REVIEW

**Comment by corivera** [comment]
Do we have any test coverage for these methods?

**Comment by danmosemsft** [comment]
Also candidates on SqlDataReader
```
       public ReadOnlyCollection<DbColumn> GetColumnSchema();
        protected internal bool IsCommandBehavior(CommandBehavior condition);
```

**Comment by danmosemsft** [comment]
Also SqlParameter implements IDbDataParameter , IDataParameter and SqlDataReader implements  IDataReader, IDataRecord in the implementation - you could choose to expose those in the ref as they're public and documented implementations on those types.

**Comment by geleems** [comment]
@corivera 
For the test cases of the 4 method being added to the contract,
* `SqlBulkCopyColumnMappingCollection.Clear()` is being used in `SqlBulkCopyColumnMappingCollectionTest.cs` `InsertAndClear_BehavesAsExpected()` test.
* `SqlBulkCopyColumnMappingCollection.RemoveAt(int index)`  is being used in `SqlBulkCopyColumnMappingCollectionTest.cs` `RemoveAt_BehavesAsExpected()` method.
* `SqlCommand.ResetCommandTimeout()` is not used anywhere, but simple enough not to add separate test case for this method. 
* `SqlParameterCollection.Add(string parameterName, System.Data.SqlDbType sqlDbType, int size, string sourceColumn)` is being used `ParametersTest.cs` `CodeCoverageSqlClient()` method.

**Comment by corivera** [comment]
@geleems Can you take a look at @danmosemsft 's comments?

**Comment by corivera** [comment]
Can you add a simple test case for ResetCommandTimeout? We'll have a hole in the test code coverage without it.

**Comment by geleems** [comment]
@corivera I made the methods used in test cases, so that they can be covered in test coverage.

File: src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public void CloneSqlCommand()
 
             SqlCommand command = connection.CreateCommand();
             command.CommandText = "select 1";
-            command.CommandTimeout = 45;
+            command.ResetCommandTimeout();
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs]: Can you put an assert here to verify that reset command timeout behaves as expected? Set the timeout value to something besides the default, then reset and check that the new value is the expected default.

Comment (corivera) [inline src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs]: The command timeout starts at 30, so it needs to be changed to something else to verify the behavior of ResetCommandTimeout. You can just keep the "command.CommandTimeout = 45;" line from before to achieve this.

File: src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs [hunk 2]
```diff
@@ -30,7 +30,8 @@ public void CloneSqlCommand()
...
+            Assert.Equal(command.CommandTimeout, 30);
```
Comment (corivera) [inline src/System.Data.SqlClient/tests/FunctionalTests/CloneTests.cs]: The command timeout starts at 30, so it needs to be changed to something else to verify the behavior of ResetCommandTimeout. You can just keep the "command.CommandTimeout = 45;" line from before to achieve this.

**Comment by danmosemsft** [comment]
Don't wait on me, I'm out.

**Comment by geleems** [comment]
@dotnet-bot test Linux x64 Release Build please

**Pull request merged**